### PR TITLE
Feature/us4 form add new words

### PR DESCRIPTION
**Pull request by ufelgen**
I created a form component according to [this US](https://github.com/ufelgen/capstone-project/issues/4) to add new vocab to my app.
I am storing my data in the local storage

### CODE REVIEW

File: components/NewWordForm/NewWordForm.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        id: nanoid(),
+        language: language[1],
+        flag: language[0],
+        translation: fields.queryLanguage1.value,
+        gender: fields.gender.value,
+      },
+    };
+
+    onCreateNew(newWord);
+    console.log("new word: ", newWord);
```
Comment (ThomasKirst) [inline components/NewWordForm/NewWordForm.js:27]: Please remove

File: helpers/rearrangeData.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export function rearrangeData(array) {
```
Comment (ThomasKirst) [inline helpers/rearrangeData.js:1]:
```suggestion
export function rearrangeData(words) {
```

File: helpers/rearrangeData.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  const allCategories = array.map((word) => word.category);
```
Comment (ThomasKirst) [inline helpers/rearrangeData.js:2]:
```suggestion
  const allCategories = words.map((word) => word.category);
```

File: helpers/rearrangeData.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  const uniqueCategories = Array.from(new Set(allCategories));
+
+  const singleCategories = uniqueCategories.map((category) => {
+    return (category = array.filter((word) => word.category === category));
```
Comment (ThomasKirst) [inline helpers/rearrangeData.js:7]:
```suggestion
    words.filter((word) => word.category === category);
```

File: components/NewWordForm/NewWordForm.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        <select name="queryLanguage" id="queryLanguage">
+          <option selected disabled>
+            language
+          </option>
+          <option value="🇭🇷-croatian" name="croatian">
+            🇭🇷
+          </option>
+          <option value="🇨🇿-czech" name="czech">
+            🇨🇿
+          </option>
```
Comment (ThomasKirst) [inline components/NewWordForm/NewWordForm.js:57]: Just as an idea: you could generate these options dynamically:

```jsx
languages.map(lang => <option value={lang.flag…} name={lang.name})
```
For that you would need an array of your languages. Not really necessary, but for your consideration ;)

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Hi Ulli,
this looks overall pretty good!
I just left a few minor remarks, especially regarding your rearrangeData function (`array` => `words`).
Keep it up :)

File: components/NewWordForm/NewWordForm.js (hunk 5 above)
Comment (ufelgen) [reply components/NewWordForm/NewWordForm.js:57]: thanks for the suggestion! yes I can and I will - when I dynamically generate my categories array in another user story! :)

**Pull request merged**